src/dragdrop.js

```diff
@@ -0,0 +1,43 @@
+let target;
+
+function routing(source, target) {
+  const savedTasks = JSON.parse(localStorage.getItem('savedTasks'));
+  if (savedTasks.length < 2) return;
+
+  const sourceA = savedTasks[source];
+  const sourceB = savedTasks[source].index;
+  let targetPointer;
+  // eslint-disable-next-line array-callback-return
+  savedTasks.map((object) => {
+    if (object.index === Number(target)) {
+      targetPointer = savedTasks.indexOf(object);
+    }
+  });
+  savedTasks[source].index = savedTasks[targetPointer].index;
+  savedTasks[targetPointer].index = sourceB;
+
+  savedTasks[source] = savedTasks[targetPointer];
+  savedTasks[targetPointer] = sourceA;
+  localStorage.setItem('savedTasks', JSON.stringify(savedTasks));
+}
```

Comment:
- [OPTIONAL] This method is called in your `dragend` event type. We have an abnormal behavior in which dragging a task in any direction(upwards or downwards), always shifts the dragged task to the start of the list even if that was not the intended action. You may consider doing some revision here.